src/components/CommentForm.jsx

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from "react";
+import { postComment } from "../api";
+
+
+function CommentForm({ user, article_id, setComments }) {
+  const [comment, setComment] = useState("");
+
+  function handleChange(event) {
+    setComment(event.target.value);
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const body = event.target[0].value;
+    if (!body) {
+      return alert("You must type somethings!")
+    }
+    postComment(article_id, body, user.username)
+      .then((newComment) => {
+        setComment("");
+        setComments((current) => {
+          return [newComment, ...current]
+        });
```

Comment:
Because this is within a then block after your api call it is not optimistically rendered.
Please change so that it is and think how to handle if the request fails also.